monitoring/prometheus/generate-prom.js

```diff
@@ -5,138 +5,100 @@ const fs = require('fs');
 const dotenv = require('dotenv');
 const AudiusLibs = require("@audius/libs");
 
-const ENVS = ["stage", "prod"]
 
+const writeFromFile = (stream, filename) => {
+  stream.write(fs.readFileSync("./ymls/" + filename))
+  stream.write("\n")
+}
 
-const yaml = (url, env) => {
-  url = url.replace("https://", "");
-  url = url.replace("http://", "");
-
-  sanatized_url = url.split(".").join("-")
+const generateJobYaml = (url, env, scheme = 'https', component = 'discovery-provider') => {
+  url = url.replace("https://", "").replace("http://", "")
+  sanitizedUrl = url.split(".").join("-")
 
   return `
-  - job_name: '${sanatized_url}'
-    scheme: https
+  - job_name: '${sanitizedUrl}'
+    scheme: '${scheme}'
     metrics_path: '/prometheus_metrics'
     static_configs:
       - targets: ['${url}']
         labels:
           host: '${url}'
           environment: '${env}'
           service: 'audius'
-          component: 'discover-provider'`
+          component: '${component}'
+`
+}
+
+const generateEnv = async (stream, env) => {
+  dotenv.config({ path: `.env.${env}`, override: true });
+
+  const ETH_REGISTRY_ADDRESS = process.env.REACT_APP_ETH_REGISTRY_ADDRESS
+  const ETH_TOKEN_ADDRESS = process.env.REACT_APP_ETH_TOKEN_ADDRESS
+  const ETH_OWNER_WALLET = process.env.REACT_APP_ETH_OWNER_WALLET
+  const ETH_PROVIDER_URL = process.env.REACT_APP_ETH_PROVIDER_URL
+
+  const ethWeb3Config = AudiusLibs.configEthWeb3(
+    ETH_TOKEN_ADDRESS,
+    ETH_REGISTRY_ADDRESS,
+    ETH_PROVIDER_URL,
+    ETH_OWNER_WALLET
+  )
+
+  const audiusLibs = new AudiusLibs({
+    ethWeb3Config,
+    isServer: true,
+    enableUserReplicaSetManagerContract: true,
+    preferHigherPatchForPrimary: true,
+    preferHigherPatchForSecondaries: true
+  })
+  await audiusLibs.init()
+  const serviceProviders = await audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderList('discovery-node');
+
+  // write a header to seperate this env from another
+  stream.write(`  ${"#".repeat(env.length)}\n`);
+  stream.write(`  ${env}\n`);
+  stream.write(`  ${"#".repeat(env.length)}\n`);
+
+  for (const sp of serviceProviders) {
+    const spEndpoint = sp.endpoint;
+    const yamlString = generateJobYaml(spEndpoint, env)
+    stream.write(yamlString);
+    stream.write("\n")
+  }
 }
 
 const main = async () => {
 
   const stream = fs.createWriteStream('prometheus.yml', { flags: 'a' });
 
-  stream.write(`global:
-  scrape_interval:     30s
-  evaluation_interval: 15s
-  # scrape_timeout is set to the global default (10s).
+  writeFromFile(stream, "base.yml")
+  writeFromFile(stream, "exporters.yml")
+  writeFromFile(stream, "exporters-audius.yml")
+  writeFromFile(stream, "load-tests.yml")
```

Comment:
I wasn't sure about adding `exporters.yml` as a default for all envs, but figured we could:

* Highly support new devs adding exporters for the first pass.
* Consider if we need `exporters-local.yml` and `exporters-prod.yml`, per say, when the situation arises.

Reply:
whats the diff btwn `exporters` and `exporters-audius`?

we could prob simplify this to:
```
ymls/
  base.yml
  stage.yaml
  prod.yaml
generateProm.js
```

and just put all hardcoded targets like load-tests and discovery-4 canary in the appropriate exporter file